descwl_coadd_task/tests/test_warps.py

```diff
@@ -0,0 +1,362 @@
+
+        result = makeWarp.run(
+            self.dataRefs,
+            skyInfo=self.skyInfo,
+            visit_summary=None,
+        )
+
+        warp = result.warp
+        mfrac = result.mfrac_warp
+        noise = result.noise0_warp
+
+        # Ensure we got an exposure out
+        self.assertIsInstance(warp, afw_image.ExposureF)
+        # Ensure that masked fraction is an ImageF object.
+        self.assertIsInstance(mfrac, afw_image.ImageF)
+        # Ensure that the noise image is a MaskedImageF object.
+        self.assertIsInstance(noise, afw_image.MaskedImageF)
+        # Ensure the warp has valid pixels
+        self.assertGreater(np.isfinite(warp.image.array.ravel()).sum(), 0)
+        # Ensure the warp has the correct WCS
+        self.assertEqual(warp.getWcs(), self.skyInfo.wcs)
+        # Ensure that mfrac has pixels between 0 and 1
+        self.assertTrue(np.nanmax(mfrac.array) <= 1)
+        self.assertTrue(np.nanmin(mfrac.array) >= 0)
+
+        coadd_bbox = self.skyInfo.bbox
+        expected_shape = (coadd_bbox.getHeight(), coadd_bbox.getWidth())
+        self.assertEqual(warp.image.array.shape, expected_shape)
+        self.assertEqual(mfrac.array.shape, expected_shape)
+        self.assertEqual(noise.image.array.shape, expected_shape)
+
+        # We didn't mask any pixels for this test, so the mask bits should be 0
+        # except where there is NO_DATA and EDGE bits set due to chip gaps.
+        bit_mask = afw_image.Mask.getPlaneBitMask(
+            "NO_DATA"
+        ) | afw_image.Mask.getPlaneBitMask("BAD")
+        # Slicing mfrac.array can consume too much memory in GitHub Actions.
+        # Multiply the matrices instead.
+        self.assertTrue((mfrac.array * (~(warp.mask.array & bit_mask)) == 0).all())
+
+    @lsst.utils.tests.methodParameters(mask_bitname=["BAD", "CR", "SAT"])
+    def test_makeWarpMfrac(self, mask_bitname):
+        """Test mfrac is being properly propagated"""
+
+        config = MakeShearWarpConfig()
+        nexpected = 4
+
+        # this masks a single pixel, which then propagates to mfrac
+        # in a set of neighboring pixels
+        for exposureRef in self.dataRefs:
+            self._fill_exposure(
+                exposureRef.get(),
+                mask_pixel=True,
+                mask_bitname=mask_bitname,
+            )
+
+        makeWarp = MakeShearWarpTask(config=config)
+        result = makeWarp.run(
+            self.dataRefs,
+            skyInfo=self.skyInfo,
+            visit_summary=None,
+        )
+
+        mfrac = result.mfrac_warp
+        warp = result.warp
+
+        wbad = np.where(mfrac.array != 0)
+        self.assertEqual(wbad[0].size, nexpected)
+
+        iflag = afw_image.Mask.getPlaneBitMask("INTRP")
+        wintrp = np.where(warp.mask.array & iflag != 0)
+        self.assertEqual(wintrp[0].size, nexpected)
+
+
+def setup_module(module):
+    lsst.utils.tests.init()
+
+
+class MatchMemoryTestCase(lsst.utils.tests.MemoryTestCase):
```

Comment:
What does this do?

Reply:
It checks for memory/resource leaks. Just a trivial subclassing of it is sufficient for the testing framework to pick it up and check for leaks.